controller.js

```diff
@@ -43,11 +44,37 @@ exports.getArticle = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  selectAllArticles().then((allArticles) => {
-    addComments(allArticles).then((allArticlesWithComments) => {
-      res.status(200).send(allArticlesWithComments);
+  const queries = Object.keys(req.query);
+
+  let sort_by = undefined;
+  let order = undefined;
+
+  if (queries.includes("sort_by")) {
+    sort_by = req.query.sort_by;
+  }
+  if (queries.includes("order")) {
+    order = req.query.order;
+  }
+
+  if (sort_by !== undefined || order !== undefined) {
+    // we are here
+
+    selectAllArticlesWithParams(sort_by, order)
```

Comment:
We shouldn't have a separate model here. You should be reusing the selectAllArticles model, but passing in the sort_by and order arguments to the parameter, then depending on whether they are there, it'll query the database in a different way. Have a think about how you can use req.query to do this.